### PR TITLE
Add 'repeat' argument to profiler.schedule

### PR DESCRIPTION
**Pull request by guyang3532**

### CODE REVIEW

File: torch/profiler/profiler.py [hunk 1]
```diff
@@ -16,16 +16,18 @@ class ProfilerAction(Enum):
...
 
 
-def schedule(*, wait: int, warmup: int, active: int) -> Callable:
+def schedule(*, wait: int, warmup: int, active: int, repeat: int = 0) -> Callable:
     """
     Returns a callable that can be used as profiler ``schedule`` argument. The profiler will wait for ``wait`` steps, then
     do the warmup for the next ``warmup`` steps, then
     do the active recording for the next ``active`` steps and then
-    repeat the cycle staring with the next step.
+    repeat the cycle starting with the next step and end after ``repeat`` cycles.
```
Comment (ilia-cher) [inline torch/profiler/profiler.py:24]: repeat the cycle starting with the next step. The number of cycles is specified by the ``repeat`` parameter. When the parameter's value is zero, the cycles will continue until the profiling is finished.

File: torch/profiler/profiler.py [hunk 2]
```diff
@@ -16,16 +16,19 @@ class ProfilerAction(Enum):
     RECORD_AND_SAVE = 3
 
 
-def schedule(*, wait: int, warmup: int, active: int) -> Callable:
+def schedule(*, wait: int, warmup: int, active: int, repeat: int = 0) -> Callable:
```
Comment (gdankel) [inline torch/profiler/profiler.py:19]: How do I get a trace collection on step 100 followed by 1100 and 2100?
I.e. I want three traces collected, the first after some workload warmup period and then a couple of more after some fixed intervals.

Comment (guyang3532) [reply torch/profiler/profiler.py:19]: You may need to write a schedule by yourself. It return ProfilerAction.WARMUP at beginning and return ProfilerAction.RECORD at step 100, 1100, 2100, and ProfilerAction.NONE at other.

Comment (gdankel) [reply torch/profiler/profiler.py:19]: I agree that a one-shot is much more useful in practice as the default. And the above use case can almost be supported with wait = 1000, repeat = 3.

**Review by gdankel — APPROVED** [review]

File: torch/profiler/profiler.py (hunk 2 above)
Comment (gdankel) [reply torch/profiler/profiler.py:19]: Right, I just question whether this API is very useful, other than as a one-shot. On the one hand you typically want to collect a trace as soon as the workload is stable and then perhaps a few times during a longer training run.
When would you use a non-zero repeat argument in practice with this API?

Comment (guyang3532) [reply torch/profiler/profiler.py:19]: I agree. A one-shot is much more useful and this api will get the one-shot result when repeat=1. If there's no this 'repeat' argument, this profiling cycles will continue until the training loop finish. Do you mean we should set repeat=1 as default? Then this api will do the one-shot profiling by default.

Comment (gdankel) [reply torch/profiler/profiler.py:19]: Right, that's what I meant yes, repeat = 1 will be more useful.

Comment (gdankel) [reply torch/profiler/profiler.py:19]: Replace repeat of 0 with 1 in my previous comments :)

Comment (guyang3532) [reply torch/profiler/profiler.py:19]: This pr has merged..We can do it in another pr...

Comment (gdankel) [reply torch/profiler/profiler.py:19]: Sure, we can discuss and refine.
